zap/src/parser/mod.rs

```diff
@@ -114,17 +114,11 @@ impl Ty {
 			Ty::U16(_) => Some(2),
 			Ty::U32(_) => Some(4),
 
-			Ty::Str { len } => {
-				if len.is_exact() {
-					Some(len.min().unwrap() as usize)
-				} else {
-					None
-				}
-			}
+			Ty::Str { len } => len.exact().map(|len| len as usize),
```

Comment:
`.map` feels like a waste for a type assertion - can you assert w/o mapping?

Reply:
This is mapping an `Option`, my bad.

Reply:
`len.exact` returns an `Option<T>`